test/metabase/upload_test.clj

```diff
@@ -539,387 +598,372 @@
+                            [2 "Millennium Falcon" " Han Solo"]]
+                           (rows-for-table table)))))))))))
     (testing "Multiple identical column names that normalize to _mb_row_id"
       (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
         (with-mysql-local-infile-on-and-off
           (mt/with-empty-db
-            (@#'upload/load-from-csv!
-             driver/*driver*
-             (mt/id)
-             "upload_test"
-             (csv-file-with ["_mb row id,ship,captain,_mb row id"
-                             "100,Serenity,Malcolm Reynolds,200"
-                             "3,Millennium Falcon, Han Solo,4"]))
-            (testing "Table and Fields exist after sync"
-              (sync/sync-database! (mt/db))
-              (let [table (t2/select-one Table :db_id (mt/id))]
-                (is (=? {:name #"(?i)upload_test"} table))
-                (testing "Check the data was uploaded into the table correctly"
-                  (is (= ["_mb_row_id", "ship", "captain"]
-                         (column-names-for-table table)))
-                  (is (= [[1 "Serenity" "Malcolm Reynolds"]
-                          [2 "Millennium Falcon" " Han Solo"]]
-                         (rows-for-table table))))))))))
+            (let [table-name (mt/random-name)]
+              (@#'upload/load-from-csv!
+               driver/*driver*
+               (mt/id)
+               table-name
+               (csv-file-with ["_mb row id,ship,captain,_mb row id"
+                               "100,Serenity,Malcolm Reynolds,200"
+                               "3,Millennium Falcon, Han Solo,4"]))
+              (testing "Table and Fields exist after sync"
+                (let [table (sync-upload-test-table! :database (mt/db) :table-name table-name)]
+                  (testing "Check the data was uploaded into the table correctly"
+                    (is (= ["_mb_row_id", "ship", "captain"]
+                           (column-names-for-table table)))
+                    (is (= [[1 "Serenity" "Malcolm Reynolds"]
+                            [2 "Millennium Falcon" " Han Solo"]]
+                           (rows-for-table table)))))))))))
     (testing "Multiple different column names that normalize to _mb_row_id"
       (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
         (with-mysql-local-infile-on-and-off
-          (mt/with-empty-db
+          (let [table-name (mt/random-name)]
+            (mt/with-empty-db
+              (@#'upload/load-from-csv!
+               driver/*driver*
+               (mt/id)
+               table-name
+               (csv-file-with ["_mb row id,ship,captain,_MB_ROW_ID"
+                               "100,Serenity,Malcolm Reynolds,200"
+                               "3,Millennium Falcon, Han Solo,4"]))
+              (testing "Table and Fields exist after sync"
+                (let [table (sync-upload-test-table! :database (mt/db) :table-name table-name)]
```

Comment:
Aside - pattern repeats a lot, and there was a lot of noise to update it. Seems like it would be worth DRYing up to simple macro. Perhaps something like:

```clojure
(with-csv-table [table]
  ["my,csv,contents"
   ...]
  ...)
```

Reply:
Yeah, I've wanted to do this for a while. I'm definitely not making that change in this PR though, it's already big enough